packages/server-wallet/src/models/channel.ts

```diff
@@ -473,25 +474,23 @@ export class Channel extends Model implements ChannelColumns {
     targetAfter: Uint256;
     targetTotal: Uint256;
   } {
-    /**
-     * The below logic assumes:
-     *  1. Each destination occurs at most once.
-     *  2. We only care about a single destination.
-     * One reason to drop (2), for instance, is to support ledger top-ups with as few state updates as possible.
-     */
     const supported = this.supported;
     if (!supported) {
-      throw new Error(`Channel passed to DriectFunder has no supported state`);
+      throw new Error(`Channel has no supported state`);
     }
-
-    const myDestination = this.participants[this.myIndex].destination;
     const {allocationItems} = checkThat(supported.outcome, isSimpleAllocation);
 
-    const myAllocationItem = _.find(allocationItems, ai => ai.destination === myDestination);
+    const myAllocationItem = this.allocationItemForParticipantIndex(this.myIndex);
     if (!myAllocationItem) {
-      throw new Error(`My destination ${myDestination} is not in allocations ${allocationItems}`);
+      throw new Error(
+        `My destination ${
+          this.participants[this.myIndex].destination
+        } is not in allocations ${allocationItems}`
+      );
```

Comment:
I don't think it's a best practice to throw an error with an unbounded data structure in it like this?

There are some options:
```
// 1. 
const err = new Error('My destination is missing from allocationItems');
logger.error({err, myDestination: this.participants[this.myIndex].destination, allocationItems});
throw err;
```

2. Extend the `Error` class with an error that can store properties.

(2) is nice, because
A. The error message is consistent. This has many benefits! (easy grepping, error analytics, etc.)
B. pino's StdErrSerializer, which invoked when you log an object with keyword `err`, will still log all the properties + the stack trace

It's also the case that the error message you're throwing is making assumptions about the implementation of `allocationItemForParticipantIndex`. So, if `allocationItemForParticipantIndex` changes, your error message might now be lying.

Reply:
Just to be clear, your suggestion 2 does not get around the fact that if `allocationItems` is included in the error, an unbonded data structure is thrown. Is that correct?

I am not sure I understand the last part of your comment. The helper method `allocationItemForParticipantIndex` fetches an allocation item (maybe multiple in the future) for a participant. How would a helper implementation change make this error invalid?

Reply:
6416aec61 uses ChannelError instead of the base Error class.

Reply:
> Just to be clear, your suggestion 2 does not get around the fact that if `allocationItems` is included in the error, an unbonded data structure is thrown. Is that correct?

That's a good point. However,
1. I think it's bad to use an _unbounded_ data structure in the error message, because humans read error messages. For instance, [graph-pino](https://www.npmjs.com/package/@graphprotocol/graph-pino) attempts to make pino logs human readable. When logging an error, it would displays `err.message`, and including `allocationItems` in `err.message` will probably make it unreadable.
2. Another negative of using _any_ data structure is that _computers_ read error messages. For instance, Sentry groups errors by message, and including a data structure in the message makes it difficult to track the grouped errors.

Reply:
Both are good points!